sklearn/tests/test_dummy.py

```diff
@@ -479,6 +480,22 @@ def test_constant_strategy_multioutput():
     _check_predict_proba(clf, X, y)
 
 
+def test_constant_strategy_y2d():
+    X = [[0], [0], [0], [0]]  # ignored
+    y = np.array([[2],
+                  [1],
+                  [2],
+                  [2]])
+
+    n_samples = len(X)
+
+    clf = DummyClassifier(strategy="constant", random_state=0,
+                          constant=[1])
+    clf.fit(X, y)
+    assert_array_equal(clf.predict(X), np.ones((n_samples, 1)))
```

Comment:
It seems the convention elsewhere is that even when trained on a column vector, predict should return a 1d array:
```

In [6]: %paste
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
for y in [[0, 1], [[0], [1]]]:
    for clf in [LogisticRegression(), RandomForestClassifier(), SVC(probability=True)]:
        print(clf.__class__.__name__, y)
        clf.fit([[0], [0]], y)
        for m in ['predict', 'predict_proba', 'decision_function']:
            try:
                print(m, getattr(clf, m)([[0], [0]]).shape)
            except AttributeError:
                pass
## -- End pasted text --
LogisticRegression [0, 1]
predict (2,)
predict_proba (2, 2)
decision_function (2,)
RandomForestClassifier [0, 1]
predict (2,)
predict_proba (2, 2)
SVC [0, 1]
predict (2,)
predict_proba (2, 2)
decision_function (2,)
LogisticRegression [[0], [1]]
/Users/joel/repos/scikit-learn/sklearn/utils/validation.py:547: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
predict (2,)
predict_proba (2, 2)
decision_function (2,)
RandomForestClassifier [[0], [1]]
/Users/joel/anaconda3/envs/scipy3k/bin/ipython:7: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  from IPython import start_ipython
predict (2,)
predict_proba (2, 2)
SVC [[0], [1]]
/Users/joel/repos/scikit-learn/sklearn/utils/validation.py:547: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
predict (2,)
predict_proba (2, 2)
decision_function (2,)
```